app/src/main/java/com/hedvig/app/feature/offer/ui/OfferModel.kt

```diff
@@ -20,15 +21,20 @@ sealed class OfferModel {
         val startDate: OfferStartDate,
         @StringRes
         val startDateLabel: Int,
-        val netMonthlyCost: MonetaryAmount,
-        val grossMonthlyCost: MonetaryAmount,
+        @StringRes
+        val checkoutTextRes: Int?,
```

Comment:
Something I've been thinking about: similar to how we're attempting to avoid data-layer objects in the android-layer, I think we should avoid android-objects in the viewmodel-layer. My reasoning is that if we ever want to attempt to re-use the viewmodel - which should be doable, given that it's more or less agnostic to the rendering-method - we would struggle to do so if we have android-specifics in the viewmodel.

Given that line of reasoning, I've been thinking that perhaps we should return enums describing what type of text we're looking to display, and then have the view-layer interpret what that means in terms of android-specific resources.

What do you think about this? @hugokallstrom @StylianosGakis

Reply:
Sounds very reasonable as an idea. How about we try it out and see how it feels.

Reply:
Yes sounds great. Ill fix!